File: frontend/src/api/inspection.ts
/**
 * inspection.ts — 区域抽查API封装
 *
 * v127 新增（多级审核体系 · 区域抽查）
 */
import apiClient from './client'

// ==================== 类型定义 ====================

export interface InspectionListItem {
  id: string
  lesson_plan_id: string
  plan_title: string
  plan_subject: string
  plan_grade: string
  author_name: string
  school_name: string
  inspector_id: string | null
  inspector_name: string
  sample_batch: string
  status: string
  status_name: string
  priority: number
  comment: string
  inspected_at: string | null
  created_at: string
}

export interface InspectionListResponse {
  items: InspectionListItem[]
  total: number
}

export interface InspectionStatsResponse {
  total_sampled: number
  total_pending: number
  total_passed: number
  total_revoked: number
  pass_rate: number
}

export interface DistrictInspectorItem {
  id: string
  inspector_id: string
  inspector_name: string
  region_id: string
  region_name: string
  created_at: string
}

// ==================== 辅助函数 ====================

function extractData<T>(resp: { data?: { data?: T } }): T {
  const d = resp?.data as Record<string, unknown> | undefined
  if (d && 'data' in d) return d.data as T
  return d as unknown as T
}

// ==================== API 函数 ====================

/** 获取抽查列表 */
export async function getInspections(params?: { status?: string; limit?: number; offset?: number }) {
  const resp = await apiClient.get('/inspections', { params })
  return extractData<InspectionListResponse>(resp)
}

/** 获取抽查详情 */
export async function getInspection(id: string) {
  const resp = await apiClient.get(`/inspections/${id}`)
  return extractData<InspectionListItem>(resp)
}

/** 提交抽查结果 */
export async function reviewInspection(id: string, req: { decision: 'passed' | 'revoked'; comment: string }) {
  const resp = await apiClient.post(`/inspections/${id}/review`, req)
  return extractData<{ message: string }>(resp)
}

/** 手动触发抽样 */
export async function batchSample(req?: { school_id?: string; sample_rate?: number }) {
  const resp = await apiClient.post('/inspections/batch-sample', req || {})
  return extractData<{ message: string; sampled_count: number }>(resp)
}

/** 分配审查员 */
export async function assignInspector(id: string, inspectorId: string) {
  const resp = await apiClient.put(`/inspections/${id}/assign`, { inspector_id: inspectorId })
  return extractData<{ message: string }>(resp)
}

/** 获取抽查统计 */
export async function getInspectionStats() {
  const resp = await apiClient.get('/inspections/stats')
  return extractData<InspectionStatsResponse>(resp)
}

/** 获取区域教研员列表 */
export async function getDistrictInspectors(regionId?: string) {
  const resp = await apiClient.get('/district-inspectors', { params: regionId ? { region_id: regionId } : {} })
  return extractData<DistrictInspectorItem[]>(resp)
}

/** 分配教研员到区域 */
export async function createDistrictInspector(req: { inspector_id: string; region_id: string }) {
  const resp = await apiClient.post('/district-inspectors', req)
  return extractData<DistrictInspectorItem>(resp)
}

/** 取消教研员分配 */
export async function deleteDistrictInspector(id: string) {
  const resp = await apiClient.delete(`/district-inspectors/${id}`)
  return extractData<{ message: string }>(resp)
}

// ==================== 状态常量 ====================

export const INSPECTION_STATUS_CONFIG: Record<string, { label: string; color: string; bg: string; icon: string }> = {
  pending:   { label: '待分配', color: '#9CA3AF', bg: 'rgba(156,163,175,0.08)', icon: '⏳' },
  assigned:  { label: '已分配', color: '#4F7BE8', bg: 'rgba(79,123,232,0.08)',  icon: '📌' },
  in_review: { label: '审查中', color: '#F59E0B', bg: 'rgba(245,158,11,0.08)',  icon: '🔍' },
  passed:    { label: '抽查通过', color: '#10B981', bg: 'rgba(16,185,129,0.08)', icon: '✅' },
  revoked:   { label: '已撤回', color: '#EF4444', bg: 'rgba(239,68,68,0.08)',   icon: '🚫' },
}

File: backend/internal/models/inspection.go
package models

// inspection.go — 区域抽查数据模型
//
// v127 新增（多级审核体系 · 区域抽查）：
//   - InspectionRecord：抽查记录实体
//   - DistrictInspectorAssignment：区域教研员管辖分配
//   - 相关常量、请求/响应结构体
//
// 对应数据库表：
//   inspection_records              — 抽查记录
//   district_inspector_assignments  — 区域教研员管辖分配

import "time"

// ==================== 抽查状态常量 ====================

const (
	InspectionStatusPending   = "pending"    // 待分配
	InspectionStatusAssigned  = "assigned"   // 已分配审查员
	InspectionStatusInReview  = "in_review"  // 审查中
	InspectionStatusPassed    = "passed"     // 抽查通过
	InspectionStatusRevoked   = "revoked"    // 发现问题，已撤回教案
)

// InspectionStatusNameMap 抽查状态中文名
var InspectionStatusNameMap = map[string]string{
	InspectionStatusPending:  "待分配",
	InspectionStatusAssigned: "已分配",
	InspectionStatusInReview: "审查中",
	InspectionStatusPassed:   "抽查通过",
	InspectionStatusRevoked:  "已撤回",
}

// ==================== 数据库实体 ====================

// InspectionRecord 抽查记录（对应 inspection_records 表）
type InspectionRecord struct {
	ID           string     `json:"id"`
	LessonPlanID string     `json:"lesson_plan_id"`
	InspectorID  *string    `json:"inspector_id"`   // 分配的教研员（可后分配）
	SampleBatch  string     `json:"sample_batch"`   // 抽样批次号
	Status       string     `json:"status"`         // pending/assigned/in_review/passed/revoked
	Priority     int        `json:"priority"`       // 0普通/1优先/2紧急
	Comment      string     `json:"comment"`        // 抽查意见
	InspectedAt  *time.Time `json:"inspected_at"`   // 完成时间
	CreatedAt    *time.Time `json:"created_at"`
}

// DistrictInspectorAssignment 区域教研员管辖分配（对应 district_inspector_assignments 表）
type DistrictInspectorAssignment struct {
	ID          string     `json:"id"`
	InspectorID string     `json:"inspector_id"`
	RegionID    string     `json:"region_id"`
	CreatedAt   *time.Time `json:"created_at"`
}

// ==================== 请求结构体 ====================

// InspectionReviewRequest 提交抽查结果请求
type InspectionReviewRequest struct {
	Decision string `json:"decision"` // passed / revoked
	Comment  string `json:"comment"`  // 抽查意见
}

// BatchSampleRequest 手动触发抽样请求
type BatchSampleRequest struct {
	SchoolID   string  `json:"school_id"`    // 可选：指定学校
	SampleRate float64 `json:"sample_rate"`  // 可选：覆盖默认抽样比例
}

// AssignInspectorRequest 分配审查员请求
type AssignInspectorRequest struct {
	InspectorID string `json:"inspector_id"`
}

// CreateDistrictInspectorRequest 分配区域教研员请求
type CreateDistrictInspectorRequest struct {
	InspectorID string `json:"inspector_id"`
	RegionID    string `json:"region_id"`
}

// ==================== 响应结构体 ====================

// InspectionListItem 抽查列表项（含教案和教研员信息）
type InspectionListItem struct {
	ID             string     `json:"id"`
	LessonPlanID   string     `json:"lesson_plan_id"`
	PlanTitle      string     `json:"plan_title"`
	PlanSubject    string     `json:"plan_subject"`
	PlanGrade      string     `json:"plan_grade"`
	AuthorName     string     `json:"author_name"`
	SchoolName     string     `json:"school_name"`
	InspectorID    *string    `json:"inspector_id"`
	InspectorName  string     `json:"inspector_name"`
	SampleBatch    string     `json:"sample_batch"`
	Status         string     `json:"status"`
	StatusName     string     `json:"status_name"`
	Priority       int        `json:"priority"`
	Comment        string     `json:"comment"`
	InspectedAt    *time.Time `json:"inspected_at"`
	CreatedAt      *time.Time `json:"created_at"`
}

// InspectionListResponse 抽查列表响应
type InspectionListResponse struct {
	Items []*InspectionListItem `json:"items"`
	Total int                   `json:"total"`
}

// InspectionStatsResponse 抽查统计响应
type InspectionStatsResponse struct {
	TotalSampled  int     `json:"total_sampled"`   // 总抽查数
	TotalPending  int     `json:"total_pending"`   // 待审查
	TotalPassed   int     `json:"total_passed"`    // 已通过
	TotalRevoked  int     `json:"total_revoked"`   // 已撤回
	PassRate      float64 `json:"pass_rate"`       // 通过率
}

// DistrictInspectorListItem 区域教研员列表项
type DistrictInspectorListItem struct {
	ID            string     `json:"id"`
	InspectorID   string     `json:"inspector_id"`
	InspectorName string     `json:"inspector_name"`
	RegionID      string     `json:"region_id"`
	RegionName    string     `json:"region_name"`
	CreatedAt     *time.Time `json:"created_at"`
}
